ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1532,9 +1532,12 @@ void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInf
           getOptions().ageAtAccessTracker.trackAgeAtAccess(serverBlobInfo.getBlobProperties().getCreationTimeInMs());
           blobData = blobAll.getBlobData();
           encryptionKey = blobAll.getBlobEncryptionKey();
-          if (encryptionKey == null) {
-            // set blobInfo only if decryption is not required. If not, maybeProcessCallbackAndComplete() will set appropriate
-            // value
+          if (encryptionKey == null || options.getChunkIdsOnly) {
```

Comment:
can u use a helper like isEncrypted() { return enc_key != null } ?

Reply:
Changed it to `Objects.isNull`. I believe that `enc_key == null` check to identify an unencrypted blob, works in this case when we are looking at the deserialized blob we just read from the server. It might not apply as a sufficient condition for a generic `isEncrypted()` in all cases. So I am trying to avoid a method like `isEncrypted` for this check.